HIRS_Utils/src/main/java/hirs/data/persist/tpm/PcrInfoShort.java

```diff
@@ -250,4 +296,18 @@ public final int hashCode() {
 
         return byteBuffer.array();
     }
+
+    private boolean isTpm1() {
+        // need to get an individual PCR and measure length to determine SHA1 v SHA 256
+        List<TPMMeasurementRecord> pcrs = this.getPcrComposite().getPcrValueList();
+        if (pcrs.size() == 0) {
+            // TODO figure out a reasonable case
+            // it's the case of an empty pcrmask, so maybe it doesn't matter
+            return false;
+        }
+
+        Digest hash = pcrs.get(0).getHash();
+        // check if the hash algorithm is SHA 1, if so it's TPM 1.2, if not it's TPM 2.0
+        return hash.getAlgorithm() == DigestAlgorithm.SHA1;
```

Comment:
So, TPM 1.2 supports SHA1 and TPM 2.0 supports SHA 1, 256, and more (especially due to algorithm agility). For our purposes, we hard coding TPM 2.0 to only report SHA 256. Because we are reusing the 1.2 data structures for appraisal, this is the simplest way of making this determination without adding additional data to the class, which was a bit of a feature requirement.

Reply:
going to mark this resolved